decidim-proposals/lib/decidim/proposals/proposal_answer_creator.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    # This class is responsible for creating the imported proposal answers
+    # and must be included in proposals component's import manifest.
+    class ProposalAnswerCreator < Decidim::Admin::Import::Creator
+      def initialize(data, context = nil)
+        @data = data
+        @context = context
+      end
+
+      # Retuns the resource class to be created with the provided data.
+      def self.resource_klass
+        Decidim::Proposals::Proposal
+      end
+
+      # Add answer to proposal
+      #
+      # Returns a proposal
+      def produce
+        resource
+      end
+
+      def finish!
+        super
+        notify(resource)
+      end
+
+      # Check if prepared resource is valid
+      #
+      # record - Decidim::Proposals::Proposal
+      #
+      # Returns true if record is valid
+      def self.resource_valid?(record)
+        return false if record.nil?
+        return false if record.errors.record.positive?
+
+        record.valid?
+      end
+
+      private
+
+      attr_reader :context
+
+      def resource
+        @resource ||= begin
+          return nil if Decidim::Proposals::Proposal.where(id: id).empty?
+
+          proposal = Decidim::Proposals::Proposal.find(id)
+          if proposal.component != component
+            proposal.errors.add(:component, :invalid)
+            return proposal
+          end
+
+          proposal.answer = answer
+          proposal.answered_at = Time.current
+          if Decidim::Proposals::Proposal::POSSIBLE_STATES.include?(state)
```

Comment:
Can we add a separate constant for `POSSIBLE_ANSWER_STATES` because we probably don't want the importer to "accidentally" e.g. withdraw proposals?

The possible answer states are `accepted`, `rejected` and `evaluating`.

Reply:
Yes